README.md

```diff
@@ -74,6 +74,8 @@ The following input parameters are supported configurations for the Endor Labs G
 |  `api_key`                            | Set to your Endor Labs API key ID |
 |  `api_secret`                         | Set to your Endor Labs API key secret |
 | `gcp_service_account`                 | Set to the GCP service account used for keyless authentication. This may not be used in conjunction with your API key   |
+| `enable_pr_comments`                  | Set to `false` to disable PR comments feature. Defaults to `true`. `ci_run` must be set to `true` and `github_token` must be provided for this |
+| `github_token`                        | Provide your github token here. This is required when `enable_pr_comments` is set to `true` |
```

Comment:
Isnt this already a requirement? The github runner already has `GITHUB_TOKEN` as a pre requisite.

Reply:
Yeah the GitHub token is already put into the runner as an environment variable by default. The question is how we get it to work with the GitHub token that is already built into GitHub Actions. We should need `permissions: pull_request: write` for this is my understanding.

Reply:
Yeah the `GITHUB_TOKEN` does get added to `secrets` and the `github` context variable. But I was not able to directly access it inside the action.
From what I found in the documentation:

`secrets` are available for use inside the workflow file and need to passed down to any actions in the workflow that need it

> To make a secret available to an action, you must set the secret as an input or environment variable in the workflow file. Review the action's README file to learn about which inputs and environment variables the action expects. For more information, see "[Workflow syntax for GitHub Actions](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsenv)."
> You can use and read secrets in a workflow file if you have access to edit the file. For more information, see "[Access permissions on GitHub](https://docs.github.com/en/get-started/learning-about-github/access-permissions-on-github)."

https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions#accessing-your-secrets


`github` context has `github.token` property available only within the execution steps of a job in the workflow file

> A token to authenticate on behalf of the GitHub App installed on your repository. This is functionally equivalent to the GITHUB_TOKEN secret. For more information, see "[Automatic token authentication](https://docs.github.com/en/actions/security-guides/automatic-token-authentication)."
Note: This context property is set by the Actions runner, and is only available within the execution steps of a job. Otherwise, the value of this property will be null.

https://docs.github.com/en/actions/learn-github-actions/contexts#github-context


An example of passing it from workflow to action available here: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#example-1-passing-the-github_token-as-an-input

Just to confirm if I understand the requirement correctly: The `GITHUB_TOKEN` is the one we need, right? And this token needs to be passed to the `endorctl` command as `--github-token=<GITHUB_TOKEN>` inside the action?

Am I missing something very obvious here?

Reply:
Also, `permissions: pull_request: write` should ideally not be needed here

> Warning: For workflows that are triggered by the pull_request_target event, the GITHUB_TOKEN is granted read/write repository permission unless the permissions key is specified and the workflow can access secrets, even when it is triggered from a fork. Although the workflow runs in the context of the base of the pull request, you should make sure that you do not check out, build, or run untrusted code from the pull request with this event. Additionally, any caches share the same scope as the base branch. To help prevent cache poisoning, you should not save the cache if there is a possibility that the cache contents were altered. For more information, see "[Keeping your GitHub Actions and workflows secure: Preventing pwn requests](https://securitylab.github.com/research/github-actions-preventing-pwn-requests)" on the GitHub Security Lab website.

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target

Reply:
pull request events and pull request target events are different. I think we are looking at pull request events.

Reply:
@rranjeet @ankurv3rma - Since the GitHub token is only available at runtime is this going to be an issue. It should just be set by default with the environment variable shouldn't it?

Reply:
@ankurv3rma It looks like we have to pass the `github-token` to `endorctl` explicitly. As long as you pass the token as a parameter, we are good. 

@IAmATeaPot418 - Looks like the action does not have access to the Github token as per the documentation. Also, I do not see an issue in passing it as an argument to the action. (except for showing the github token in plain text, is there a way to avoid that in runners?)

Reply:
@ankurv3rma - With respect to permissions, we need to have `issues: write` and `pull-requests:write` are required. I have tried to read/write/update Pull Request and Issue Comments without these permissions and the requests have failed. 

I am not sure how we set the permissions in the Github token that is generated.But based on this documentation, https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token the permissions are already present.

Reply:
The `GITHUB_TOKEN` which is given in the action input is being passed to `endorctl`. And the runner always masks the values for these secrets and doesn't show them in plain text.

I think it would make sense to always have users set the `issues: write` and `pull-requests:write` permissions explicitly instead of depending on them being set automatically, just to be sure that they'll always be present.